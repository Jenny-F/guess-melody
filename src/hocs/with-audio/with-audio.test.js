import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withAudio from "./with-audio.js";

const mockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

mockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

const MockComponentWrapped = withAudio(mockComponent);

it(`Should WithAudio render correctly`, () => {
  const tree = renderer.create(
      <MockComponentWrapped
        isPlaying={false}
        src={``}
        onPlayButtonClick={() => {}}
      />, {
        createNodeMock() {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
