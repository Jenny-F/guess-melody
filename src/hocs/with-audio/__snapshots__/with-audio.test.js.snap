// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should WithAudio render correctly 1`] = `
<div>
  <audio />
</div>
`;
